src/zombiereborn.cpp

```diff
@@ -76,6 +79,298 @@ CON_ZR_CVAR(zr_infect_spawn_mz_min_count, "Minimum amount of Mother Zombies to b
+				if (bTeamDefault)
+					m_vecHumanDefaultClass.AddToTail(pHumanClass);
+				
+				pHumanClass->PrintInfo();
+			}
+			else 
+			{
+				ZRZombieClass *pZombieClass;
+				if (pszBase)
+				{
+					ZRZombieClass *pBaseZombieClass = GetZombieClass(pszBase);
+					if (pBaseZombieClass)
+					{
+						pZombieClass = new ZRZombieClass(pBaseZombieClass);
+						pZombieClass->Override(pSubKey);
+					}
+					else
+					{
+						Warning("Could not find specified base \"%s\" for %s!!!", pszBase, pszClassName);
+						continue;
+					}
+				}
+				else
+					pZombieClass = new ZRZombieClass(pSubKey);
+
+				m_ZombieClassMap.Insert(hash_32_fnv1a_const(pSubKey->GetName()), pZombieClass);
+				if (pSubKey->GetBool("team_default", false))
+					m_vecZombieDefaultClass.AddToTail(pZombieClass);
+				
+				pZombieClass->PrintInfo();
+			}
+		}
+	}
+}
+void CZRPlayerClassManager::ApplyBaseClass(ZRClass* pClass, CCSPlayerPawn *pPawn)
+{
+	pPawn->m_iMaxHealth = pClass->iHealth;
+	pPawn->m_iHealth = pClass->iHealth;
+	pPawn->SetModel(pClass->szModelPath.c_str());
+	pPawn->m_flVelocityModifier = pClass->flSpeed;
+	pPawn->m_flGravityScale = pClass->flGravity;
+}
+
+ZRHumanClass* CZRPlayerClassManager::GetHumanClass(const char *pszClassName)
+{
+	uint16 index = m_HumanClassMap.Find(hash_32_fnv1a_const(pszClassName));
+	if (!m_HumanClassMap.IsValidIndex(index))
+		return nullptr;
+	return m_HumanClassMap[index];
+}
+
+void CZRPlayerClassManager::ApplyHumanClass(ZRHumanClass *pClass, CCSPlayerPawn *pPawn)
+{
+	ApplyBaseClass(pClass, pPawn);
+	CCSPlayerController *pController = CCSPlayerController::FromPawn(pPawn);
+	if (pController)
+		CZRRegenTimer::StopRegen(pController);
+}
+
+void CZRPlayerClassManager::ApplyDefaultHumanClass(CCSPlayerPawn *pPawn)
+{
+	if (m_vecHumanDefaultClass.Count() == 0)
+	{
+		Warning("Missing default human class!!!");
+		return;
+	}
+	ApplyHumanClass(m_vecHumanDefaultClass[rand() % m_vecHumanDefaultClass.Count()], pPawn);
+}
+
+ZRZombieClass* CZRPlayerClassManager::GetZombieClass(const char *pszClassName)
+{
+	uint16 index = m_ZombieClassMap.Find(hash_32_fnv1a_const(pszClassName));
+	if (!m_ZombieClassMap.IsValidIndex(index))
+		return nullptr;
+	return m_ZombieClassMap[index];
+}
+
+void CZRPlayerClassManager::ApplyZombieClass(ZRZombieClass *pClass, CCSPlayerPawn *pPawn)
+{
+	ApplyBaseClass(pClass, pPawn);
+	CCSPlayerController *pController = CCSPlayerController::FromPawn(pPawn);
+	if (pController)
+		CZRRegenTimer::StartRegen(pClass->flHealthRegenInterval, pClass->iHealthRegenCount, pController);
+}
+
+void CZRPlayerClassManager::ApplyDefaultZombieClass(CCSPlayerPawn *pPawn)
+{
+	if (m_vecZombieDefaultClass.Count() == 0)
+	{
+		Warning("Missing default zombie class!!!");
+		return;
+	}
+	ApplyZombieClass(m_vecZombieDefaultClass[rand() % m_vecZombieDefaultClass.Count()], pPawn);
+}
+
+int CZRRegenTimer::s_iRegenTimerCount = 0;
+int CZRRegenTimer::s_vecRegenTimersIndex[MAXPLAYERS];
+CZRRegenTimer *CZRRegenTimer::s_vecRegenTimers[MAXPLAYERS];
```

Comment:
They are already declared static in the header file. `s_vecRegenTimersIndex` is being used because not all 64 players will have a timer. The timer in `s_vecRegenTimers` can have their position changed whenever a new regen timer is inserted(infected) or removed(cured).